tuning/utils/preprocessing_utils.py

```diff
@@ -33,12 +34,59 @@
 JSON_OUTPUT_KEY = "output"
 
 
+# check if the provided dataset is pretokenized or not
+# the check is taken from trl
+# https://github.com/huggingface/trl/blob/ddf4c8dc3ecf6d9ee2b24f94c62182ffd682c808/trl/trainer/sft_trainer.py#L498-L509
+def is_pretokenized_dataset(data: Union[str, Dataset, IterableDataset]):
+    if not data:
+        return False
+    if isinstance(data, str):
+        try:
+            data = datasets.load_dataset("json", data_files=data, split="train[:1]")
+        except DatasetGenerationError as e:
+            raise DatasetGenerationError("failed to load the provided dataset") from e
+
+    return ("input_ids" in data.column_names) and ("labels" in data.column_names)
+
+
 def validate_data_args(data_args: configs.DataArguments, packing: bool):
 
     assert isinstance(
         data_args.training_data_path, str
     ), "Training data path has to be set and str"
 
+    is_train_data_pretokenized = is_pretokenized_dataset(data_args.training_data_path)
+    is_eval_data_pretokenized = is_pretokenized_dataset(data_args.validation_data_path)
```

Comment:
this is going to return False when validation file not set. later in line 77, it will return valueerror .

validation file is optional from top level API. Only when not None, these checks need to happen. Please add unit test

Reply:
oh I see you added `if data_args.validation_data_path` later on, thats fine then